rollbar/client.go

```diff
@@ -0,0 +1,149 @@
+package rollbar
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+const apiURL = "https://api.rollbar.com/api/1"
+
+// Client represents the rollbar client.
+type Client struct {
+	APIKey      string
+	APIScheme   string
+	APIHost     string
+	APIBasePath string
+}
+
+// Option adds the base url and other parameters to the client.
+type Option func(*Client) error
+
+// NewClient is a constructor.
+func NewClient(apiKey string, opts ...Option) (*Client, error) {
```

Comment:
I don't see that we're using opts anywhere in our use case: https://github.com/babbel/terraform-provider-rollbar/blob/61a7cc8c5924b564aa5ab98f13c6857519ca9fe6/rollbar/provider.go#L37
Did I just missed them or here we're willing to be generic? Or it's only for testing purpose? If it's only for testing, I find it's quite confusing then. We're introducing this attribute here, which is fine, but is generic, and also below introducing a method `parseOptions`, which's purpose from what I see is just to apply this baseUrl to the client (basically modify it). 
Want to emphasize that I might be wrong and don't understand the use case. However, if I'm not, then I'd suggest to not try to be overgeneric, but simply have a baseUrl as a param to the builder function.

Reply:
@graffzon This options pattern is sometimes used. It was already in the original implementation and I prefer to make the PR bigger that it is now.